src/m3u8/m3u8-parser.js

```diff
@@ -436,7 +440,8 @@
               // setup an encryption key for upcoming segments
               key = {
                 method: entry.attributes.METHOD || 'AES-128',
-                uri: entry.attributes.URI
+                uri: entry.attributes.URI,
+                iv: entry.attributes.IV || null
```

Comment:
How would you have a Uint32Array in json then ? I just made changes untill it passed all your automated tests, change it to fit your style however you like.